crates/fluvio/src/producer/accumulator.rs

```diff
@@ -137,20 +149,53 @@ impl RecordAccumulator {
         let mut batch = ProducerBatch::new(self.batch_size, self.compression);
 
         match batch.push_record(record) {
-            Some(push_record) => {
+            Ok(push_record) => {
                 batch_events.notify_new_batch().await;
-
                 if batch.is_full() {
                     batch_events.notify_batch_full().await;
                 }
-
                 batches.push_back(batch);
+                Ok(PushRecord::new(
+                    push_record.into_future_record_metadata(partition_id),
+                ))
+            }
+            Err((size, record)) => {
+                // send the single record without batch size limit
+                self.create_batch_single_record(
+                    size,
+                    record,
+                    &mut batches,
+                    batch_events,
+                    partition_id,
+                )
+                .await
+            }
+        }
+    }
 
+    pub async fn create_batch_single_record(
+        &self,
+        batch_size: usize,
+        record: Record,
+        batches: &mut VecDeque<ProducerBatch>,
+        batch_events: &BatchEvents,
+        partition_id: PartitionId,
+    ) -> Result<PushRecord, ProducerError> {
+        let mut batch = ProducerBatch::new(batch_size, self.compression);
+        match batch.push_record(record) {
+            Ok(push_record) => {
+                batch_events.notify_new_batch().await;
+                batches.push_back(batch);
                 Ok(PushRecord::new(
                     push_record.into_future_record_metadata(partition_id),
                 ))
             }
-            None => Err(ProducerError::RecordTooLarge(self.batch_size)),
+            Err((size, _record)) => {
+                // This should never happen, as we are creating a batch with a single record, with
+                // the same size as the record
+                // Only if we add a max_size option to the record (not batch)
+                Err(ProducerError::RecordTooLarge(size))
```

Comment:
Instead of checking record size which doesn't matter, we should migrated to batch size concept

Reply:
I don't understand.
Let say that my `batch_size` is 50000, and I try to send a record with 50001 of size, what should happen?

For example, I am almost sure that in Kafka, this case will just send directly (not using batch mechanism).
Now, on fluvio, we will have an error.

Is it correct?

I propose to create a batch with a single record for these cases.

Reply:
in that case, it should reject it same as in the Kafka which as https://kafka.apache.org/documentation/

Reply:
Strictly Uniform Sticky Partitioner is not deprecated?